src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -232,14 +244,8 @@
                           "maximum number of borrows (%s)")
                      (:id instance)
                      max-borrows)
-          (try
-            (flush-instance-fn pool instance)
-            (finally
-              (.releaseItem pool instance false))))
-        (.releaseItem pool instance)))
-    ;; if we get here, it was from a borrow and we got a Retry, so we just
-    ;; return it to the pool.
-    (.releaseItem (:pool instance) instance)))
+          (flush-instance-fn instance))
+        (.releaseItem pool instance)))))
```

Comment:
Yeah, what @jpinsonault said sounds right.  We had the `.releaseItem` overload in the past because the underlying Java concurrency construct that we we were using for locking needed to be decremented each time we were done with borrowing an instance (to avoid a deadlock later on) - even though we didn't end up actually returning the instance back to the pool because we wanted to dispose of it.  That became irrelevant in the later JRubyPool implementation because we didn't have a need to decrement a lock counter when disposing of an item anymore.  We only kept the no-op overload at the time for backward compatibility in the API.  I can't exactly remember why now, but I think I suggested to @jpinsonault that he go ahead and remove the unnecessary overload as part of this PR because it seemed to simplify some other part of the rework he was doing.